frontend/javascripts/admin/task/task_create_form_view.js

```diff
@@ -68,46 +70,88 @@ type State = {
 };
 
 export function handleTaskCreationResponse(responses: Array<TaskCreationResponse>) {
+  if (responses.length === 0) {
+    return;
+  }
   const successfulTasks = [];
   const failedTasks = [];
+  let teamName = null;
 
   responses.forEach((response: TaskCreationResponse, i: number) => {
     if (response.status === 200 && response.success) {
-      successfulTasks.push(
-        `${response.success.id},${response.success.creationInfo ||
-          "null"},(${response.success.editPosition.join(",")}) \n`,
-      );
+      if (!teamName) {
+        teamName = response.success.team;
+      }
+      successfulTasks.push(response.success);
     } else if (response.error) {
       failedTasks.push(`Line ${i}: ${response.error} \n`);
     }
   });
-
-  Modal.info({
-    title: `${successfulTasks.length} tasks were successfully created. ${
-      failedTasks.length
-    } tasks failed.`,
-    content: (
-      <div>
-        {successfulTasks.length > 0 ? (
-          <div>
-            Successful Tasks:
-            <pre>
-              taskId,filename,position
-              <br />
-              {successfulTasks}
-            </pre>
-          </div>
-        ) : null}
-        {failedTasks.length > 0 ? (
+  if (successfulTasks.length > 0) {
+    downloadTasksAsCSV(successfulTasks);
+  }
+  if (failedTasks.length > 0) {
+    Modal.info({
+      title: `Failed to create ${failedTasks.length} tasks.`,
+      content: (
+        <div>
           <div>
             Failed Tasks:
             <pre>{failedTasks}</pre>
           </div>
-        ) : null}
-      </div>
-    ),
-    width: 600,
-  });
+        </div>
+      ),
+      width: 600,
+    });
+  }
+}
+
+export function taskToText(task: APITask) {
+  const {
+    id,
+    created,
+    dataSet,
+    type,
+    neededExperience,
+    editPosition,
+    editRotation,
+    status,
+    team,
+    boundingBoxVec6,
+    projectName,
+    script,
+  } = task;
+  const neededExperienceAsString = `${neededExperience.domain},${neededExperience.value}`;
+  const [editPositionAsString, editRotationAsString] = [editPosition, editRotation].map(
+    position => `${position[0]},${position[1]},${position[2]}`,
+  );
+  const totalNumberOfInstances = status.open + status.active + status.finished;
+  const boundingBoxAsString = boundingBoxVec6
+    ? boundingBoxVec6.reduce((partialString, val) => `${partialString}${val},`)
+    : "_,_,_,_,_,_,";
+  const scriptId = script ? script.id : "_";
+
+  const taskAsString =
+    `${id},${created},${dataSet},${type.id},${neededExperienceAsString},${editPositionAsString},` +
+    `${editRotationAsString},${totalNumberOfInstances},${team},${boundingBoxAsString}${projectName},${scriptId}`;
+  return taskAsString;
+}
+
+export function downloadTasksAsCSV(tasks: Array<APITask>) {
+  if (tasks.length < 0) {
```

Comment:
length < 0? This cannot happen, I think ^^

Reply:
This might have happened when the method was called by the task list view with no tasks selected. Now I added a check in the code of the task list view and only call this method if the tasks array is not empty. :+1:

Reply:
And the compare should have being `length <= 0` :sweat_smile: